lib/components/RangePicker.vue

```diff
@@ -0,0 +1,326 @@
+    startBoundStyle(): { left: string } {
+      return { left: `${this.start * 100}% ` }
+    },
+    endBoundStyle(): { left: string } {
+      return { left: `${this.end * 100}%` }
+    },
+    classList(): { [key: string]: boolean } {
+      return {
+        [`range-picker--${this.variant}`]: !!this.variant,
+        'range-picker--rounded': this.rounded
+      }
+    }
+  },
+  watch: {
+    value([start, end]) {
+      this.start = start
+      this.end = end
+    }
+  },
+  beforeMount() {
+    library.add(faGripLinesVertical)
+  },
+  methods: {
+    snapValue(value: number): number {
+      return round(value / this.snap) * this.snap
+    },
+    rangeWidth(): number {
+      return this.$el?.querySelector('.range-picker__bounds')?.getBoundingClientRect().width ?? 0
+    },
+    dragStartBound(dx: number) {
+      const newValue = this.snapValue(dx / this.rangeWidth())
+      // Ensure start value doesn't get too close to end value
+      if (newValue < this.end - this.minDistance) {
+        this.start = round(newValue, this.precision)
+        /**
+         * Update the values of the range (both start and end)
+         * @event update
+         * @param Number[] New value of the range
+         */
+        this.$emit('update', [this.start, this.end])
+      }
+    },
+    dragEndBound(dx: number) {
+      const newValue = this.snapValue(dx / this.rangeWidth())
+      // Ensure end value doesn't get too close to start value
+      if (newValue > this.start + this.minDistance) {
+        this.end = round(newValue, this.precision)
+        /**
+         * Update the values of the range (both start and end)
+         * @event update
+         * @param Number[] New value of the range
+         */
+        this.$emit('update', [this.start, this.end])
+      }
+    },
+    dragBounds(dx: number) {
+      const diff = this.snapValue(this.end - this.start)
+      const newValue = this.snapValue(dx / this.rangeWidth())
+
+      this.start = round(newValue, this.precision)
+      this.end = round(newValue + diff, this.precision)
+      /**
+       * Update the values of the range (both start and end)
+       * @event update
+       * @param Number[] New value of the range
+       */
+      this.$emit('update', [this.start, this.end])
+    },
+    valueWithUnit(value: number | string): string {
+      return typeof value === 'number' ? `${value}px` : `${value}`
+    }
+  }
+})
+</script>
+
+<template>
+  <div class="range-picker" :class="classList">
+    <slot />
+    <div class="range-picker__bounds" :style="boundsStyle">
+      <div v-draggable.relative class="range-picker__bounds__overlay" :style="overlayStyle" @dragged="dragBounds"></div>
+      <button v-draggable :style="startBoundStyle" class="range-picker__bounds__start btn" @dragged="dragStartBound">
+        <fa icon="fa-grip-lines-vertical" fixed-width />
+      </button>
+      <button v-draggable class="range-picker__bounds__end btn" :style="endBoundStyle" @dragged="dragEndBound">
```

Comment:
On the button (start and end), I suggest to change the cursor to cursor: 'col-resize' (and keep the cursor as it while dragging.

Reply:
Changing, thanks for the suggestions!